pinot-controller/src/test/java/com/linkedin/pinot/controller/api/restlet/resources/SegmentCompletionProtocolDeserTest.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.controller.api.restlet.resources;
+
+import com.linkedin.pinot.common.protocols.SegmentCompletionProtocol;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class SegmentCompletionProtocolDeserTest {
+  @Test
+  public void testSerializeRequestParams() {
+    // Test with all parameters
+    SegmentCompletionProtocol.Request.Params reqParams = new SegmentCompletionProtocol.Request.Params()
+        .withOffset(1)
+        .withBuildTimeMillis(123L)
+        .withInstanceId("instance")
+        .withNumRows(23)
+        .withReason("toomanyrows")
+        .withSegmentLocation("file.tmp")
+        .withExtraTimeSec(5)
+        .withSegmentName("name")
+        .withWaitTimeMillis(123L);
+
+    SegmentCompletionProtocol.SegmentCommitRequest segmentCommitRequest = new SegmentCompletionProtocol.SegmentCommitRequest(reqParams);
+    Assert.assertEquals(
+        "http://hostport/segmentCommit?name=name&offset=1&instance=instance&reason=toomanyrows&buildTimeMillis=123&waitTimeMillis=123&extraTimeSec=5&rowCount=23&location=file.tmp"
+        , segmentCommitRequest.getUrl("hostport"));
+
+    // Test without segment location
+    reqParams = new SegmentCompletionProtocol.Request.Params()
+        .withOffset(1)
+        .withBuildTimeMillis(123L)
+        .withInstanceId("instance")
+        .withNumRows(23)
+        .withReason("toomanyrows")
+        .withExtraTimeSec(5)
+        .withSegmentName("name")
+        .withWaitTimeMillis(123L);
+
+    segmentCommitRequest = new SegmentCompletionProtocol.SegmentCommitRequest(reqParams);
+    segmentCommitRequest.getUrl("hostport");
+    Assert.assertEquals(
+        "http://hostport/segmentCommit?name=name&offset=1&instance=instance&reason=toomanyrows&buildTimeMillis=123&waitTimeMillis=123&extraTimeSec=5&rowCount=23"
+        , segmentCommitRequest.getUrl("hostport"));
```

Comment:
you should add code to decode this URL, get the parts and make sure that each part is parsed correctly into request params using the method to decode params.